packages/core/src/agent/MessageSender.ts

```diff
@@ -272,20 +273,20 @@ export class MessageSender {
     const allServices = this.transportService.findDidCommServices(connection)
 
     //Separate queue service out
-    const services = allServices.filter((s) => !isDidCommTransportQueue(s.serviceEndpoint))
+    let services = allServices.filter((s) => !isDidCommTransportQueue(s.serviceEndpoint))
     const queueService = allServices.find((s) => isDidCommTransportQueue(s.serviceEndpoint))
 
     //If restrictive will remove services not listed in schemes list
     if (transportPriority?.restrictive) {
-      services.filter((service) => {
+      services = services.filter((service) => {
```

Comment:
@JamesKEbert @burdettadam The services were not being reassigned meaning the sort and filter did not do anything. This should fix it.

This means that currently it will take the http endpoint, send the ping and receive the ping response after which the http endpoint is closed 😄 

So if we could merge this rather sooner than later that would be great!

Reply:
How embarrassing! Thanks for catching that.

Reply:
Whoooops--my bad!
As a side note, the `array.sort()` method does actually do its operation in place and returns too, so it doesn't necessarily have to be reassigned too, but doesn't hurt to reassign it. :)

Reply:
Ah you're right. So weird that filter does not do it's operation in place, but sort does. Thanks